api/v1/configurationpolicy_types.go

```diff
@@ -211,120 +175,134 @@ const (
-type ConfigurationPolicyStatus struct {
-	ComplianceState   ComplianceState  `json:"compliant,omitempty"`         // Compliant/NonCompliant/UnknownCompliancy
-	CompliancyDetails []TemplateStatus `json:"compliancyDetails,omitempty"` // reason for non-compliancy
-	// An ISO-8601 timestamp of the last time the policy was evaluated
-	LastEvaluated string `json:"lastEvaluated,omitempty"`
-	// The generation of the ConfigurationPolicy object when it was last evaluated
-	LastEvaluatedGeneration int64 `json:"lastEvaluatedGeneration,omitempty"`
-	// List of resources processed by the policy
-	RelatedObjects []RelatedObject `json:"relatedObjects,omitempty"`
-}
+// ObjectTemplate describes how an object should look
+type ObjectTemplate struct {
+	ComplianceType         ComplianceType         `json:"complianceType"`
+	MetadataComplianceType MetadataComplianceType `json:"metadataComplianceType,omitempty"`
 
-// CompliancePerClusterStatus contains aggregate status of other policies in cluster
-type CompliancePerClusterStatus struct {
-	AggregatePolicyStatus map[string]*ConfigurationPolicyStatus `json:"aggregatePoliciesStatus,omitempty"`
-	ComplianceState       ComplianceState                       `json:"compliant,omitempty"`
-	ClusterName           string                                `json:"clustername,omitempty"`
+	// ObjectDefinition defines required fields, in whole or in part, to be compared with objects on
+	// the cluster.
+	//
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
+
+	// RecordDiff specifies whether (and where) to log the diff between the object on the cluster and
+	// the objectDefinition in the policy. Options are "Log" to log the diff in the
+	// config-policy-controller pod and "None" to not log the diff. Defaults to "None".
+	RecordDiff RecordDiff `json:"recordDiff,omitempty"`
 }
 
-// ComplianceMap map to hold CompliancePerClusterStatus objects
-type ComplianceMap map[string]*CompliancePerClusterStatus
+// ConfigurationPolicySpec defines the desired configuration of objects on the cluster along with
+// how the configuration policy should handle when the cluster doesn't match the configuration
+// policy.
+type ConfigurationPolicySpec struct {
+	Severity           Severity           `json:"severity,omitempty"`
+	RemediationAction  RemediationAction  `json:"remediationAction"`
+	EvaluationInterval EvaluationInterval `json:"evaluationInterval,omitempty"`
+	// +kubebuilder:default:=None
+	PruneObjectBehavior PruneObjectBehavior `json:"pruneObjectBehavior,omitempty"`
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
-//+kubebuilder:printcolumn:name="Compliance state",type="string",JSONPath=".status.compliant"
+	// NamespaceSelector defines the list of namespaces to include/exclude for objects defined in
+	// spec.object-templates. All selector rules are ANDed. If 'include' is not provided but
```

Comment:
Nitpick: Since there is a hyphen, this is how you reference it with JSON syntax.  :)
```suggestion
	// spec["object-templates"]. All selector rules are ANDed. If 'include' is not provided but
```